test/runtests.jl

```diff
@@ -86,3 +86,73 @@ end
   @test nls_model.nls_meta.nequ == 256 * 256
   @test all(0 .<= nls_model.meta.x0 .<= 1)
 end
+
+@testset "SVM-Train" begin
+  nlp_train,nls_train,sol = svm_train_model()
+  @test typeof(nlp_train) <: FirstOrderModel
+  @test typeof(nls_train) <: FirstOrderNLSModel
+  @test typeof(sol) == Vector{Int64}
+
+
+  x = nlp_train.meta.x0
+  f = obj(nlp_train, x)
+  F = residual(nls_train, x)
+  @test f ≈ dot(F, F) / 2
+
+  g = grad(nlp_train, x)
+  JtF = jtprod_residual(nls_train, x, F)
+  @test all(g .≈ JtF)
```

Comment:
You can simplify all this by calling
```julia
test_well_defined(model, nls_model, sol)
test_objectives(model, nls_model)
```
Is the solution really a vector of Ints??? It should probably be converted to a vector of Floats, no?

Reply:
Yes they are binary labels, i.e. {-1, 1}. Why would they be floats?

Reply:
I test that x0 and sol have the same data type in `test_well_defined()`. Normally we don't solve the problem in integer variables. Maybe don't call `test_well_defined()` then, in this case.

Reply:
Yeah it's weird because there's no "solution" in a sense (i.e. it's not F(x) - b). Ok I'll remove my changes (they were only local).